core/src/main/java/io/jenkins/pluginhealth/scoring/probes/AbstractJavaImportsCheckProbe.java

```diff
@@ -0,0 +1,93 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * The abstract probe checks for imports in {@code .java} files in a plugin.
+ */
+public abstract class AbstractJavaImportsCheckProbe extends Probe {
+    public static final int ORDER = SCMLinkValidationProbe.ORDER + 100;
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractJavaImportsCheckProbe.class);
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Path scmRepository = context.getScmRepository();
+
+        /* The "maxDepth" is set to Integer.MAX_VALUE because a repository can have multiple modules with class files in it. We do not want to miss any ".java" file.  */
+        try (Stream<Path> paths = Files.find(scmRepository, Integer.MAX_VALUE, (path, $) ->
```

Comment:
Here `paths` is `javaFiles` right? Or `javaFilesPath`?

The difference to me with something like `(path, $) ->` that you see at the end of the line is that at the end of the line `path` is indeed a path, and we don't know what it is exactly, we just know it's a path.

But the `Stream` that we create definitely contains all the Java files of the plugin, so we can have a more explicit name than just `paths`, because we know what kind of path that is.